src/main/java/john/parser/Parser.java

```diff
@@ -0,0 +1,125 @@
+package john.parser;
+
+import java.util.regex.Pattern;
+
+import john.commands.ByeCommand;
+import john.commands.Command;
+import john.commands.DeadlineCommand;
+import john.commands.DeleteCommand;
+import john.commands.EventCommand;
+import john.commands.FindCommand;
+import john.commands.IncorrectCommand;
+import john.commands.ListCommand;
+import john.commands.MarkCommand;
+import john.commands.TodoCommand;
+import john.commands.UnmarkCommand;
+
+
+/**
+ * Represents a parser to parse user input.
+ */
+public class Parser {
+    private static final String REGEX_DEADLINE =
+            "[^ ](.*) /by ([1-9]|[0-2][0-9]|(3)[0-1])/(((0)?[0-9])|((1)[0-2]))/[0-9]{4}"
+                    + "(( ([01][0-9]|2[0-3])([0-5][0-9]))|)";
+    private static final String REGEX_EVENT =
+            "[^ ](.*) /at ([1-9]|[0-2][0-9]|(3)[0-1])/(((0)?[0-9])|((1)[0-2]))/[0-9]{4}"
+                    + "(( ([01][0-9]|2[0-3])([0-5][0-9]))|)";
+    private static final String REGEX_LIST =
+            "(([1-9]|[0-2][0-9]|(3)[0-1])/(((0)?[0-9])|((1)[0-2]))/[0-9]{4}|)";
+
+    /**
+     * Parses the user input, and returns the corresponding command.
+     * @param input The user input.
+     * @return Command representing the user input.
+     */
+    public Command parseCommand(String input) {
+        String mainCommand = input.split(" ", 2)[0].toLowerCase();
+        String params = parseParams(input);
+
+        switch (mainCommand) {
+        case TodoCommand.COMMAND_WORD:
+            return prepareTodo(params);
+        case DeadlineCommand.COMMAND_WORD:
+            return prepareDeadline(params);
+        case EventCommand.COMMAND_WORD:
+            return prepareEvent(params);
+        case ListCommand.COMMAND_WORD:
+            return prepareList(params);
+        case MarkCommand.COMMAND_WORD:
+            return prepareMark(params);
+        case UnmarkCommand.COMMAND_WORD:
+            return prepareUnmark(params);
+        case DeleteCommand.COMMAND_WORD:
+            return prepareDelete(params);
+        case FindCommand.COMMAND_WORD:
+            return prepareFind(params);
+        case ByeCommand.COMMAND_WORD:
+            return new ByeCommand();
+        default:
+            return new IncorrectCommand(input);
+        }
+    }
```

Comment:
Perhaps you could leave a blank line between methods to improve readability?

Reply:
Hey @LJXSean, good catches, I've gone ahead to fix them!